packages/core/src/language-server/code-action-provider.ts

```diff
@@ -0,0 +1,194 @@
+        let version = parseInt(this.documents.getDocumentVersion(filePath));
+        let snapshot = TextDocument.create(
+          filePath,
+          'ts',
+          version,
+          this.documents.getDocumentContents(filePath)
+        );
+
+        return TextDocumentEdit.create(
+          OptionalVersionedTextDocumentIdentifier.create(snapshot.uri, version),
+          change.textChanges.map((edit) => {
+            let original = this.transformManager.getOriginalRange(
+              change.fileName,
+              edit.span.start,
+              edit.span.start + edit.span.length
+            );
+
+            let start = offsetToPosition(
+              this.documents.getDocumentContents(uriToFilePath(filePath)),
+              original.originalStart
+            );
+            let end = offsetToPosition(
+              this.documents.getDocumentContents(uriToFilePath(filePath)),
+              original.originalEnd
+            );
+
+            // We need to re-write \@ts-ignore directives for embedded templates
+            // Failing to do so would replace the problematics code with \@ts-ignore
+            // instead of prepending it with \@glint-ignore.
+            if (
+              shouldReWriteIgnoreJsDiagnostic(
+                fix.fixName,
+                edit.newText,
+                original.mapping?.sourceNode
+              )
+            ) {
+              return this.insertGlintIgnore(filePath, edit, start);
+            }
+
+            return TextEdit.replace(
+              {
+                start,
+                end,
+              },
+              edit.newText
+            );
+          })
+        );
+      });
+
+      return CodeAction.create(fix.description, { documentChanges }, CodeActionKind.QuickFix);
+    });
+  }
+
+  // This mimics what happens in TS/JS but for when we are in an embedded template context.
+  // We fix up the indenting because this is the same behavior that occurs what inserting
+  // \@ts-ignore checks
+  private insertGlintIgnore(filePath: string, edit: ts.TextChange, start: Position): TextEdit {
+    edit.newText = '{{! @glint-ignore }}\n';
+
+    let linesOfNewText = edit.newText.split('\n');
+
+    if (/^[ \t]*$/.test(linesOfNewText[linesOfNewText.length - 1])) {
+      let contents = this.documents.getDocumentContents(uriToFilePath(filePath)).split('\n')[
+        start.line
+      ];
+      let indent = /^[ |\t]+/.exec(contents)?.[0] ?? '';
+      linesOfNewText[linesOfNewText.length - 1] = indent;
+    }
+
+    return TextEdit.insert(start, linesOfNewText.join('\n'));
+  }
+
+  private cleanDiagnosticCode(diagnostics: Diagnostic[]): number[] {
+    return diagnostics
+      .map((diag) => {
+        if (diag.code) {
+          return typeof diag.code === 'string' ? parseInt(diag.code) : diag.code;
+        } else if (diag.source) {
+          return parseInt(diag.source.replace('glint:ts(', '').replace(')', ''));
```

Comment:
So these are diagnostics coming directly from the LSP bindings. I don't know where we would track them since every time a CodeAction is invoked they will be new.